N24DCCN161/buoi5/bai10.cpp

```diff
@@ -0,0 +1,24 @@
+#include <iostream>
+using namespace std;
+int main(){
+    int n;
+    cin >> n;
+    int d;
+    for(int i = 2; i <= n; i++){
```

Comment:
Vòng for là vòng lặp sử dụng trong trường hợp biết trước số lần lặp. For này em quy định chạy từ 2 tới n, nhưng n lại bị thay đổi bên trong vòng lặp (thành ra vòng for sẽ chạy mà không biết chính xác số lần lặp). Điều này không đúng ý nghĩa của for lắm. Sửa lại thành while (sử dụng while trong trường hợp này có thể xử lý thêm, bỏ một vài trường hợp thừa, làm chương trình tối ưu hơn, suy nghĩ nha)